src/Aspire.Hosting.Azure/Provisioning/Internal/DefaultArmClientProvider.cs

```diff
@@ -108,6 +108,19 @@ public async Task<IEnumerable<ISubscriptionResource>> GetAvailableSubscriptionsA
             return locations.OrderBy(l => l.DisplayName);
         }
 
+        public async Task<IEnumerable<string>> GetAvailableResourceGroupsAsync(string subscriptionId, CancellationToken cancellationToken = default)
+        {
+            var subscription = await armClient.GetSubscriptions().GetAsync(subscriptionId, cancellationToken).ConfigureAwait(false);
```

Comment:
In many cases we would have already queried for the subscription ID or gotten the subscription list, can we cache the result or do we want to fetch each time.